sync/class.jetpack-sync-module-posts.php

```diff
@@ -320,43 +317,52 @@ function filter_post_content_and_add_links( $post_object ) {
 
 	public function save_published( $new_status, $old_status, $post ) {
 		if ( 'publish' === $new_status && 'publish' !== $old_status ) {
-			$this->just_published[] = $post->ID;
+			$this->just_published[ $post->ID ] = true;
 		}
 
-		if ( 'trash' === $new_status && 'trash' !== $old_status ) {
-			$this->just_trashed[] = $post->ID;
-		}
+		$this->previous_status[ $post->ID ] = $old_status;
 	}
 
 	public function wp_insert_post( $post_ID, $post = null, $update = null ) {
 		if ( ! is_numeric( $post_ID ) || is_null( $post ) ) {
 			return;
 		}
 
-		if ( Jetpack_Constants::get_constant( 'DOING_AUTOSAVE' ) ) {
-			$is_auto_save = true;
-		} else {
-			$is_auto_save = false;
-		}
-
-		if ( ! in_array( $post_ID, $this->just_published ) ) {
-			$just_published = false;
-		} else {
-			$just_published = true;
-		}
-
 		// workaround for https://github.com/woocommerce/woocommerce/issues/18007
 		if ( $post && 'shop_order' === $post->post_type ) {
 			$post = get_post( $post_ID );
 		}
 
-		call_user_func( $this->action_handler, $post_ID, $post, $update, $is_auto_save, $just_published );
+		$previous_status = isset( $this->previous_status[ $post_ID ] ) ?
+			$this->previous_status[ $post_ID ] :
+			self::DEFAULT_PREVIOUS_STATE;
+
+		$just_published = isset( $this->just_published[ $post_ID ] ) ?
+			$this->just_published[ $post_ID ] :
+			false;
+
+		$state = array(
+			'is_auto_save' => (bool) Jetpack_Constants::get_constant( 'DOING_AUTOSAVE' ),
+			'previous_status' => $previous_status,
+			'just_published' => $just_published
+		);
+		/**
+		 * Filter that is used to add to the post flags ( meta data ) when a post gets published
+		 *
+		 * @since 5.8.0
```

Comment:
Could you add the `@module sync` text to you Sync hooks, so it gets categorized by the parser?
https://developer.jetpack.com/module/sync/

Thank you!

Reply:
Will do. I had no idea we had such magic ❤️